src/Platforms/MySQL/CollationMetadataProvider/ConnectionCollationMetadataProvider.php

```diff
@@ -8,16 +8,34 @@
 use Doctrine\DBAL\Exception;
 use Doctrine\DBAL\Platforms\MySQL\CollationMetadataProvider;
 
+use function str_starts_with;
+use function substr;
+
 /** @internal */
 final class ConnectionCollationMetadataProvider implements CollationMetadataProvider
 {
-    public function __construct(private readonly Connection $connection)
+    public function __construct(private readonly Connection $connection, private bool $useUtf8mb3)
     {
     }
 
+    public function normalizeCollation(string $collation): string
+    {
+        if ($this->useUtf8mb3 && str_starts_with($collation, 'utf8_')) {
+            return 'utf8mb3' . substr($collation, 4);
```

Comment:
I'd make things more obvious here:

```suggestion
            return 'utf8mb3' . substr($collation, strlen('utf8'));
```